client/src/pages/CalendarPage/CalendarPage.js

```diff
@@ -31,24 +33,55 @@ const dailyTaskDateFormatter = (date) => {
   } ${date.getDate()}, ${date.getFullYear()}`;
 };
 
+// Convert the task data fetched from the server into JS objects of the
+// {title, locationText, startTimeText, endTimeText, isOngoing}
+const convertTaskData = (fetchedTaskData) => {
+  return [];
+};
+
+const padSingleDigit = (num) => {
+  return num <= 9 ? `0${num}` : num.toString();
+};
+
+const buildDailyTaskRoute = (currDate) => {
+  // getMonth returns 0-11
+  const currMonthStr = padSingleDigit(currDate.getMonth() + 1);
+  // getDate returns 1-31
+  const currDayStr = padSingleDigit(currDate.getDate());
+  return (
+    "/api/tasks/day/" +
+    currDate.getFullYear().toString() +
+    currMonthStr +
+    currDayStr
+  );
+};
+
 const CalendarPage = () => {
   const [showModal, setModal] = useState(false);
 
   const toggleModal = () => {
     setModal(!showModal);
   };
   const [currDate, setCurrDate] = useState(new Date());
-  const sampleTask = {
-    title: "gdsc meeting",
-    location: "dh2020",
-    startTime: "05:00pm",
-    endTime: "07:00pm",
-    isOngoing: false,
-  };
-  const sampleTasks = [];
-  for (let i = 0; i < 15; i++) {
-    sampleTasks.push({ ...sampleTask });
-  }
+  const [taskData, setTaskData] = useState([]);
+  const [showError, setShowError] = useState(false);
+
+  useEffect(() => {
+    const getTasks = async () => {
+      await instance
+        .get(buildDailyTaskRoute(currDate))
+        .then((taskData) => {
+          console.log(taskData);
+          setTaskData(convertTaskData(taskData));
```

Comment:
I would suggest only setting the taskData and then from there, you should use another function to actually do the conversion to the stuff that you'll actually need